spec/mini_scheduler/manager_spec.rb

```diff
@@ -195,6 +195,33 @@ def perform
       manager.stop!
     end
 
+    def queued_jobs(manager, with_hostname:)
+      hostname = with_hostname ? manager.hostname : nil
+      key = MiniScheduler::Manager.queue_key(manager.queue, hostname)
+      redis.zrange(key, 0, -1).map(&:constantize)
+    end
+
+    it 'should recover from Redis flush' do
+      manager = MiniScheduler::Manager.new(enable_stats: false)
+      manager.ensure_schedule!(Testing::SuperLongJob)
+      manager.ensure_schedule!(Testing::PerHostJob)
+
+      expect(queued_jobs(manager, with_hostname: false)).to include(Testing::SuperLongJob)
+      expect(queued_jobs(manager, with_hostname: true)).to include(Testing::PerHostJob)
+
+      redis.flushdb
```

Comment:
flushdb kind of makes me nervous, is there any way you can be more surgical here? Maybe if all our keys are prefixed just delete keys with that prefix? do we flushdb much in our specs?

Reply:
I think we flush a couple of times in Discourse core. Anyway, I pushed a new version which deletes only keys with the `_scheduler_` prefix.